server/rest/src/main/java/org/infinispan/rest/NettyRestServer.java

```diff
@@ -70,30 +49,55 @@ private static EmbeddedCacheManager createCacheManager(String cfgFile) {
       }
    }
 
-   private NettyRestServer(EmbeddedCacheManager cacheManager, RestServerConfiguration configuration, NettyJaxrsServer netty,
-                           Consumer<? super EmbeddedCacheManager> onStop) {
+   private NettyRestServer(EmbeddedCacheManager cacheManager, RestServerConfiguration configuration) {
       this.cacheManager = cacheManager;
       this.configuration = configuration;
-      this.netty = netty;
-      this.onStop = onStop;
    }
 
    @Override
    public void start() {
-      netty.start();
-      ResteasyDeployment deployment = netty.getDeployment();
       configuration.getIgnoredCaches().forEach(this::ignoreCache);
       RestCacheManager restCacheManager = new RestCacheManager(cacheManager, this::isCacheIgnored);
-      Server server = new Server(configuration, restCacheManager);
-      deployment.getRegistry().addSingletonResource(server);
-      deployment.getProviderFactory().register(new RestAccessLoggingHandler(), ContainerRequestFilter.class,
-            ContainerResponseFilter.class);
-      log.startRestServer(configuration.host(), configuration.port());
+      server = Optional.of(new Server(configuration, restCacheManager));
```

Comment:
That's a good point. I though about usecases when someone calls `stop(); stop();`. That that's obviously wrong and besides... those are our internals... Fixed.